authentication/auth.go

```diff
@@ -527,16 +509,32 @@ func GenerateTokenFromDefaultsV2(schemaURL string, accountServiceURL string, url
 	if claims["survey_metadata"] != nil {
 		surveyMetadata = claims["survey_metadata"].(map[string]interface{})
 	}
-	for key, value := range data {
-		updatedData[key] = value
-	}
 
 	initialData := surveyMetadata["data"].(map[string]interface{})
 
 	for key, value := range initialData {
 		updatedData[key] = value
 	}
 
+	/*
+		This method below is used to add any metadata (that is a parameter) which is intended to be of type boolean to
+		requiredSchemaMetadata for the typing in "updatedData" to be correct. Without this method, a parameter added to
+		the URL set as a boolean for example, "&example=True", would be interpreted as a String hence the type in the
+		token would be incorrect. It doesn't apply to other types because those do not need to be updated/changed in
+		"updatedData". It's essential to ensure the values in "updatedData" are of the correct type for the types in
+		token to be correct.
+	*/
+	requiredSchemaMetadata = addAdditionalMetadata(surveyMetadata, requiredSchemaMetadata)
```

Comment:
Is this only for quick launch? Does the method need to be renamed if it is only handling bools to make it clearer what is being done?

Reply:
This is only for quick launch. A more appropriate name could be `addUrlBooleanMetadata`?

Reply:
Updated in new PR